configure.ac

```diff
@@ -56,92 +61,42 @@ AC_SEARCH_LIBS([dlopen], [dl], [], [AC_MSG_ERROR([NCCL OFI Plugin requires dlope
+      [AC_MSG_WARN([Disabling tests due to Neuron configuration.])
+       enable_tests=no],
+      [test "${enable_tests}" = "yes" -a "${found_mpi}" = "no"],
+      [AC_MSG_ERROR([Tests requested, but MPI not found.  Aborting.])])
+AM_CONDITIONAL([ENABLE_TESTS], [test "$enable_tests" != "no"])
+
+# Enable output at the TRACE level for unit tests.
+AC_ARG_ENABLE([trace],
+   [AS_HELP_STRING([--enable-trace], [Enable printing trace messages in test programs])])
 AC_MSG_CHECKING([whether to enable trace messages])
-AS_IF([test "x${enable_trace}" = "xyes" ],
+AS_IF([test "${enable_trace}" = "yes" ],
       [trace=1
        AC_MSG_RESULT(yes)],
-       [trace=0
+      [trace=0
        AC_MSG_RESULT(no)])
-AC_DEFINE_UNQUOTED([OFI_NCCL_TRACE], [${trace}], [Defined to 1 if aws-ofi-nccl was configured with --enable-trace, 0 otherwise])
-
-dnl Check for LTTNG tracing capability
-AC_ARG_WITH([lttng],
-    AS_HELP_STRING([--with-lttng=DIR],
-           [Enable tracing capability with LTTNG @<:@default=no@:>@]))
-AS_IF([test -n "$with_lttng"],
-      [
-       AC_CHECK_LIB(lttng-ust,lttng_ust_probe_register, [], AC_MSG_ERROR("lttng-ust not found"))
-       AS_IF([test -d $withval/lib64], [lttng_libdir="lib64"], [lttng_libdir="lib"])
-       LDFLAGS="-L$withval/$lttng_libdir $LDFLAGS"
-       AC_DEFINE(HAVE_LTTNG, 1, [Build with LTTNG tracing])
-      ],
-      [
-       AC_DEFINE(HAVE_LTTNG, 0, [Build without LTTNG tracing])
-      ]
-)
-
-AS_IF([test "x$with_cuda" != xno],
-  [
-    AS_IF([test "x$with_cuda" != xcheck], [cuda_path=$with_cuda], [cuda_path=/usr/local/cuda])
-    AC_MSG_CHECKING([CUDA library path])
-    AS_IF([test -d $cuda_path/lib64], [cuda_libdir="lib64"], [cuda_libdir="lib"])
-    CFLAGS="-I$cuda_path/include $CFLAGS"
-    LDFLAGS="-L$cuda_path/$cuda_libdir $LDFLAGS"
-    AC_CHECK_HEADERS([cuda_runtime.h],
-      [AC_DEFINE([HAVE_CUDA], [1], [Define if CUDA is available])],
-      [AS_IF([test "x$with_cuda" != xcheck],
-        [AC_MSG_ERROR([unable to find cuda_runtime.h])],
-        [with_cuda=no
-         AC_DEFINE([HAVE_CUDA], [0], [Define if CUDA is available])])])
-  ],
-  [AC_DEFINE([HAVE_CUDA], [0], [Define if CUDA is available])])
-
-AM_CONDITIONAL(HAVE_CUDA, test "x$with_cuda" != xno)
-
-AS_IF([test "x$enable_neuron" != xno],
-    AS_IF([test "x$with_cuda" != xno], [AC_MSG_ERROR([Cuda (--with_cuda) and Neuron devices (--enable_neuron) cannot be defined at same time])])
-    [AC_DEFINE([HAVE_NEURON], [1], [Define if Neuron support is available])],
-    [AC_DEFINE([HAVE_NEURON], [0], [Define if Neuron support is available])])
-
-AM_CONDITIONAL(ENABLE_NEURON, test "x$enable_neuron" != xno)
-
-AC_CHECK_HEADERS([rdma/fabric.h], [],[
-	AC_MSG_ERROR([unable to find required headers])
-])
-
-AC_SEARCH_LIBS([fi_getinfo], [fabric], [], [
-  AC_MSG_ERROR([unable to find libfabric])
-])
+AC_DEFINE_UNQUOTED([OFI_NCCL_TRACE], [${trace}], [Defined to 1 unit test output should include TRACE level])
```

Comment:
It isn't just unit tests which uses trace so why call it out?

Reply:
It's only the unit tests.  The plugin emits TRACE logs regardless of this constant.  All this constant does is decide whether or not the TRACE logs will be printed when running the unit tests (since the unit tests provide their own logger).